### DISCUSSION TITLE
Why is this line text red and not blue?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#6572] [Help] [answered]
**Reproduction URL**

https://github.com/frederikhors/iss-tailwind-priority-color

**Describe your issue**

Steps:

1. `git clone https://github.com/frederikhors/iss-tailwind-priority-color`
1. `npm install`
1. `npm run dev`
1. open `http://localhost:3000`

The line should be blue, not red.

The final html is:

```html
<span class="text-red-500 text-blue-600">I should be BLUE!</span>
```

![image](https://user-images.githubusercontent.com/41120635/146457305-af075ddf-957b-4fb0-871c-0f2e36f8a56d.png)

**Comment by RobinMalfait** [comment, marked as answer]
If 2 selectors have the same specificity then the order matters inside the css file:

```css
.a { color: green }
.b { color: red }
```

The order in your html doesn't matter.
```html
<div class="a b">I will be red</div>
<div class="b a">I will also be red</div>
```

It's not a good idea to rely on these competing selectors. Instead render the correct class conditionally.

E.g.: In React you can do something like this:
```js
<div className={condition ? 'text-red-500' : 'text-blue-600'}></div>
```

**Reply by frederikhors** [reply]
I don't understand, @RobinMalfait.

If I postpone a class it should prevail, right?

In the play it works: https://play.tailwindcss.com/n5cKAusg53

Why not here?

**Reply by frederikhors** [reply]
I need to add a different color based on tailwind class, not on a prop.

**Reply by frederikhors** [reply]
In what order are the tailwind `text-*` classes?

**Reply by RobinMalfait** [reply]
All colors in Tailwind are considered the same "level", therefore the first we find will be generated first.

The order of the classes in your html doesn't matter, that's not something we can fix in Tailwind.
You should not rely on on this behaviour for your own code.
E.g.: https://play.tailwindcss.com/8YnkN4Z4EU

Passing a `className` prop to a component if that component already contains classes is generally a bad idea. More info on that topic:
- https://gist.github.com/RobinMalfait/7cf7a0498039027a1d591f5f2b908a95
- https://gist.github.com/RobinMalfait/490a0560a7cfde985d435ad93f8094c5

So instead of doing this:
https://github.com/frederikhors/iss-tailwind-priority-color/blob/43bfb7c63a9f661fd025e8cc35825d7a3aec5f1e/src/lib/Span.svelte#L5
Passthrough a `variant` (which you can default to a certain value if you want to), and if you need to override it then you can "change" the variant, not append to existing classNames.

**Reply by frederikhors** [reply]
You're right. I understood. Thank you very much, @RobinMalfait, really, thanks.
